src/components/Publish/Navigation/index.tsx

```diff
@@ -13,7 +15,8 @@ export default function Navigation(): ReactElement {
   }: FormikContextType<FormPublishData> = useFormikContext()
 
   function handleStepClick(step: number) {
-    setFieldValue('user.stepCurrent', step)
+    // Change route to include steps
+    router.push({ pathname: `${router.pathname}`, query: { step: step } })
```

Comment:
why do we need to set both, a pathname and a query as part of options object? As this now looks like we have 2 systems to get `step` across. Was thinking that with `publish/[step]` system we can simply do `router.push(`/publish/${step}`)` so no need to keep track of additional `query.step`, no?

Reply:
yeah, sure. I'm not sure there's any difference as I was catching router.pathname `publish/[step]` and changing his [step] prop. But it's alright, your solution seems cleaner 👍🏽